src/micromamba/install.cpp

```diff
@@ -28,88 +30,75 @@ init_install_parser(CLI::App* subcom)
-
-    subcom->add_flag("--extra-safety-checks",
-                     create_options.extra_safety_checks,
-                     "Perform extra safety checks (compute SHA256 sum for each file)");
-    subcom->add_option("--safety-checks",
-                       create_options.safety_checks,
-                       "Set safety check mode: enable, disable or warn");
+                     always_copy.set_cli_config(0),
+                     always_copy.description());
+
+    auto& extra_safety_checks = config.at("extra_safety_checks").get_wrapped<bool>();
+    subcom->add_flag("--extra-safety-checks,!--no-extra-safety-checks",
+                     extra_safety_checks.set_cli_config(0),
+                     extra_safety_checks.description());
+
+    auto& safety_checks = config.at("safety_checks").get_wrapped<VerificationLevel>();
+    subcom->add_set("--safety-checks",
+                    safety_checks.set_cli_config(""),
+                    { "enabled", "warn", "disabled" },
+                    safety_checks.description());
 }
 
-void
-load_install_options(Context& ctx)
-{
-    load_general_options(ctx);
-    load_prefix_options(ctx);
-    load_rc_options(ctx);
-    load_network_options(ctx);
-    load_channel_options(ctx);
-
-    SET_BOOLEAN_FLAG(strict_channel_priority);
-    SET_BOOLEAN_FLAG(allow_softlinks);
-    SET_BOOLEAN_FLAG(always_softlink);
-    SET_BOOLEAN_FLAG(always_copy);
-
-    if (ctx.always_softlink && ctx.always_copy)
-    {
-        throw std::runtime_error("'always-softlink' and 'always-copy' options are exclusive.");
-    }
-
-    if (!create_options.safety_checks.empty())
-    {
-        if (to_lower(create_options.safety_checks) == "disabled")
-        {
-            ctx.safety_checks = VerificationLevel::DISABLED;
-        }
-        else if (to_lower(create_options.safety_checks) == "warn")
-        {
-            ctx.safety_checks = VerificationLevel::WARN;
-        }
-        else if (to_lower(create_options.safety_checks) == "enabled")
-        {
-            ctx.safety_checks = VerificationLevel::ENABLED;
-        }
-        else
-        {
-            LOG_ERROR << "Could not parse option for --extra-safety-checks\n"
-                      << "Select none (default), warn or fail";
-        }
-    }
-
-    SET_BOOLEAN_FLAG(extra_safety_checks);
-}
 
 void
 set_install_command(CLI::App* subcom)
 {
     init_install_parser(subcom);
 
     subcom->callback([&]() {
-        auto& ctx = Context::instance();
+        auto& configuration = Configuration::instance();
 
         parse_file_options();
-        load_install_options(ctx);
+        load_configuration();
 
-        if (!create_options.specs.empty() && !ctx.target_prefix.empty())
+        auto& specs = configuration.at("specs").value<std::vector<std::string>>();
+
+        if (!specs.empty())
         {
-            install_specs(create_options.specs, false);
+            check_target_prefix(true, true, true, true);
```

Comment:
here it would be interesting to use boolean flags:

`check_target_prefix(MM_CREATE | MM_INSTALL)...`

Reply:
or multiple named functions that call to a `check_target_prefix_impl`

like `check_target_prefix_install`